core/src/main/java/org/apache/iceberg/AllManifestsTable.java

```diff
@@ -234,12 +225,14 @@ public long start() {
 
     @Override
     public long length() {
-      return manifestListTask.length();
+      // return a generic length to avoid looking up the actual length
+      return 8192;
     }
 
     @Override
     public Expression residual() {
-      return manifestListTask.residual();
+      // this table is unpartitioned so the residual is always constant
+      return residual;
```

Comment:
Is this ever used?  I wonder can we just return ResidualEvaluator.unpartitioned(Expressions.alwaysFalse()) or something

Reply:
I don't think this is used, but I would still make it the correct residual. An unpartitioned table always has the same residual as the original filter, so this is correct.